perun-core/src/main/java/cz/metacentrum/perun/core/entry/GroupsManagerEntry.java

```diff
@@ -1535,8 +1283,8 @@ public Group createGroupUnion(PerunSession sess, Group resultGroup, Group operan
 		}
 
 		// Authorization
-		if ((!AuthzResolver.isAuthorized(sess, Role.VOADMIN, resultGroup) && !AuthzResolver.isAuthorized(sess, Role.GROUPADMIN, resultGroup)) ||
-				(!AuthzResolver.isAuthorized(sess, Role.VOADMIN, operandGroup) && !AuthzResolver.isAuthorized(sess, Role.GROUPADMIN, operandGroup))) {
+		if ((!AuthzResolver.authorizedInternal(sess, "createGroupUnion_Group_Group_policy", Collections.singletonList(resultGroup))) ||
+				(!AuthzResolver.authorizedInternal(sess, "createGroupUnion_Group_Group_policy", Collections.singletonList(operandGroup)))) {
```

Comment:
It should be possible to merge these two calls into a single one.

Reply:
Done